_docs/concepts/encryption.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: guide
+title: StorageOS Docs - Encryption
+anchor: concepts
+module: concepts/encryption
+---
+
+# Encryption
+
+Encryption is enabled on a per volume basis using the
+`storageos.com/encryption` label (for more information see [Encrypted
+Volumes](/docs/operations/encrypted-volumes)).
+
+## Encryption Keys
+
+StorageOS encrypts volumes on disk using AES-256 in XTS-AES mode with 512 bit
+keys as recomended by [ NIST](
+https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38e.pdf
+), with encryption keys being derived using [ HKDF](
+https://eprint.iacr.org/2010/264.pdf ). 
+
+Keys and Initialisation vectors are generated using the [ crypto/rand ]( https://godoc.org/crypto/rand ) package.
+
+| Key                   | Size                 | Usage                                                     |
+| :-------------------- | :------------------- | :-------------------------------------------------------- |
+| Volume                | 512 bits random data | Used by XTS-AES to encrypt and decrypt disk blocks        |
+| User                  | 256 bits random data | Used to encrypt Volume keys                               |
+| Initialisation Vector | 256 bits random data | Used as 'salt' when encrypting volume keys with user keys |
+
+Encryption keys are created per volume. However user keys are created per
+namespace. This means that volumes in the same namespace have their respective
+volume master key encrypted with the same user key but with different
+initialisation vectors.
+
+The components required to derive the encryption key are stored in a Kubernetes
+secret. By default these secrets are stored in the namespace that StorageOS is
+installed into. As Kubernetes secrets are only base64 encoded, it is recommend
+to [encrypt secrets at
+rest.](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) 
```

Comment:
Worth mentioning Vault here specifically I think, since it comes up as a roadmap item so often. Can we come up with a howto guide that we can point to?